emulator/src/kreator/preprocessor.rs

```diff
@@ -310,29 +309,25 @@ fn handle_macro_locals(code: &Vec<String>) -> Result<Vec<String>, &'static str>
     Ok(handled_code)
 }
 
-fn generate_label_name(taken_names: &Vec<String>) -> String {
+fn generate_label_name(taken_names: &Vec<String>) -> Result<String, &'static str> {
     let mut new_label = String::new();
-    let mut rng = rand::thread_rng();
-    
-    while taken_names.contains(&new_label) || new_label.is_empty() {
-        new_label = get_random_char().to_string();
-        for _ in 0..4 {
-            if rng.gen_bool(0.5) {
-                new_label = format!("{}{}", new_label, get_random_char());
-            } else {
-                new_label = format!("{}{}", new_label, get_random_number());
+    let mut label_base = 'A';
+    let mut label_count = 0;
+
+    new_label = format!("{}{}", label_base, label_count);
+    while taken_names.contains(&new_label) {
```

Comment:
Maybe im missing something but isn't taken_names.len() automatically the next number? Or rather `taken_names.len()/10000` is the letter and `taken_names.len()%10000` the number?

Reply:
You are partially correct, as this is implemented poorly by me. In case of the generation method yes, the 2 snippets you provided would work. However the way `generate_label_name()` is called slightly differs from the expected: it does not receive all _generated_ names, instead it receives a list of _all_ label names so far, be they generated or not.

In action this would mean that if labels exist before or between macros containing labels the enumeration could look something like this: `GN1, GN5, GN6`. However I think your point is very valid and we should trade the slightly different behaviour for a very potent performance increase, especially in cases of larger programs.

As far as your comment goes I would assume you would second me on that and prefer your proposed implementation?

Reply:
I would prefer that yes. On top of that, if you feel like it, you could have some counter that just counts the amount of "global" labels and just subtract it.

We still need to check if the label is available before actually using it though (cause someone could just name a label "GN1" or something) in that case I guess we would just increment by one and subtract from that globals_counter.

Do you agree?

Reply:
Sounds like the best solution to this, I will implement it asap (probably tomorrow) and afterwards the PR should be ready to merge :)

Reply:
I have adjusted the issue with the latest commits, although I'm only partially happy with the solution. Take a look at it and tell me if it needs more work or if we can merge it like that.

The reason behind a variable being passed back and forth is that in cases where I tried to implement it with a static variable I experienced some really bad side effects making the tests pass 50% of the time which I was unable to fix. Maybe I am unaware of some Rust mechanic but this seemed like the simplest solution for now.